tests/test_client.py

```diff
@@ -119,3 +129,53 @@ def test_result_list_if_empty():
 
     with pytest.raises(StopIteration):
         next(results)
+
+
+@pytest.mark.parametrize(
+    "options_arg",
+    [
+        {"headers": {"Content-Type": "application/json;charset=UTF-8"}},
+        {"headers": {"random-header": "nice random"}},
+        {},
+    ],
+    ids=["overwrite", "new", "none"],
+)
+def test_headers_unclobbered_update(options_arg, no_fields):
```

Comment:
its a matter of preference, i dont like parameterized tests too much certainly if you change your test code to differ per parameter.

its opening up for weirdness during refactoring.
does it hurt to split up in 2 or more tests? so the test code doesnt need to change regarding which parameter is passed?